### PR TITLE
Add onChanged calls to YustFilePicker and YustImagePicker

### PR DESCRIPTION
**Pull request by maxsk-ger**

### CODE REVIEW

File: lib/src/widgets/yust_file_picker.dart [hunk 1]
```diff
@@ -299,6 +303,9 @@ class YustFilePickerState extends State<YustFilePicker> {
       allowMultiple: widget.allowMultiple,
     );
     if (result != null) {
+      if (_fileHandler.getFiles().isEmpty) {
+        widget.onInitialUpload?.call();
+      }
```
Comment (Alorcus) [inline lib/src/widgets/yust_file_picker.dart:308]: This logic should be in _uploadFile()

File: lib/src/widgets/yust_image_picker.dart [hunk 2]
```diff
@@ -354,7 +358,9 @@ class YustImagePickerState extends State<YustImagePicker> {
             // than maxHeight/-Width
             imageQuality: quality,
           );
-
+          if (_fileHandler.getFiles().isEmpty) {
+            widget.onInitialUpload?.call();
+          }
```
Comment (Alorcus) [inline lib/src/widgets/yust_image_picker.dart:363]: Same here

File: lib/src/widgets/yust_image_picker.dart [hunk 3]
```diff
@@ -372,6 +378,9 @@ class YustImagePickerState extends State<YustImagePicker> {
               // than maxHeight/-Width
               imageQuality: quality);
           if (image != null) {
+            if (_fileHandler.getFiles().isEmpty) {
+              widget.onInitialUpload?.call();
+            }
```
Comment (Alorcus) [inline lib/src/widgets/yust_image_picker.dart:383]: Same here

File: lib/src/widgets/yust_image_picker.dart [hunk 4]
```diff
@@ -384,6 +393,9 @@ class YustImagePickerState extends State<YustImagePicker> {
       }
       // Else, we are on Web
       else {
+        if (_fileHandler.getFiles().isEmpty) {
+          widget.onInitialUpload?.call();
+        }
```
Comment (Alorcus) [inline lib/src/widgets/yust_image_picker.dart:398]: Same here

**Review by Alorcus — CHANGES_REQUESTED** [review]
This does not work for file picker. It is not possible to upload files:
https://user-images.githubusercontent.com/63072611/232129640-3b1b69af-cee1-4c22-8896-f3e2fe85cfc4.mp4

When working on file upload logic, please test full offline mode every time (using an external mobile that is offline or disconnect your computer from the Internet). The behavior in full offline mode is very different from normal mode.

File: lib/src/widgets/yust_file_picker.dart (hunk 1 above)
Comment (maxsk-ger) [reply lib/src/widgets/yust_file_picker.dart]: The function has now been deleted and replaced by an onChanged-call.

File: lib/src/widgets/yust_file_picker.dart [hunk 5]
```diff
@@ -340,6 +341,8 @@ class YustFilePickerState extends State<YustFilePicker> {
     _processing[newYustFile.name] = false;
     if (!newYustFile.cached) {
       widget.onChanged!(_fileHandler.getOnlineFiles());
+    } else {
+      widget.onChanged!(_fileHandler.getFiles().toList());
```
Comment (Alorcus) [inline lib/src/widgets/yust_file_picker.dart:345]: This behavior seems odd. We should always return the same kind of files.

File: lib/src/widgets/yust_image_picker.dart [hunk 6]
```diff
@@ -451,6 +451,8 @@ class YustImagePickerState extends State<YustImagePicker> {
     }
     if (!newYustFile.cached) {
       widget.onChanged!(_fileHandler.getOnlineFiles());
+    }else {
+      widget.onChanged!(_fileHandler.getFiles().toList());
```
Comment (Alorcus) [inline lib/src/widgets/yust_image_picker.dart:455]: Same here

File: lib/src/widgets/yust_image_picker.dart [hunk 7]
```diff
@@ -41,6 +41,7 @@ class YustImagePicker extends StatefulWidget {
   /// default is 15
   final int imageCount;
 
+
```
Comment (Alorcus) [inline lib/src/widgets/yust_image_picker.dart:44]: Please be careful not to commit more changes than necessary

**Review by Alorcus — CHANGES_REQUESTED** [review]

**Review by Alorcus — APPROVED** [review]

**Review by janneskoehler — APPROVED** [review]

**Pull request merged**